src/virtual_dom/vtag.rs

```diff
@@ -406,7 +410,7 @@ impl VDiff for VTag {
         let (reform, mut ancestor) = {
             match ancestor {
                 Some(VNode::VTag(mut vtag)) => {
-                    if self.tag == vtag.tag {
+                    if self.tag == vtag.tag && self.key == vtag.key {
```

Comment:
This seems unnecessary, why did you add this check?

Reply:
I think this was done to detect if the key was changed and treat it as a new node.  It was part of the code @deep-gauva had already.

Reply:
I see, we should reuse the node if the tag is the same

Reply:
Can you please remove the `key` check?